pkg/controller/master-controller-manager/kc-status-controller/controller.go

```diff
@@ -0,0 +1,120 @@
+	"k8s.io/client-go/tools/record"
+	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	// ControllerName is the name of this controller.
+	ControllerName = "kkp-kc-status-controller"
+)
+
+// Reconciler watches the Kubermatic Configuration status and updates the Kubermatic Edition and Kubermatic Version.
+type Reconciler struct {
+	ctrlruntimeclient.Client
+	log      *zap.SugaredLogger
+	recorder record.EventRecorder
+	versions kubermatic.Versions
+}
+
+// Add creates a new Kubermatic Configuration status controller and sets up watches.
+func Add(
+	ctx context.Context,
+	mgr manager.Manager,
+	numWorkers int,
+	log *zap.SugaredLogger,
+	namespace string,
+	versions kubermatic.Versions,
+) error {
+	reconciler := &Reconciler{
+		Client:   mgr.GetClient(),
+		recorder: mgr.GetEventRecorderFor(ControllerName),
+		log:      log.Named(ControllerName),
+		versions: versions,
+	}
+
+	ctrlOptions := controller.Options{Reconciler: reconciler, MaxConcurrentReconciles: numWorkers}
+	c, err := controller.New(ControllerName, mgr, ctrlOptions)
+	if err != nil {
+		return err
+	}
+
+	nsPredicate := predicate.ByNamespace(namespace)
+
+	// watch the Kubermatic Configuration in the given namespace
+	if err := c.Watch(&source.Kind{Type: &kubermaticv1.KubermaticConfiguration{}},
+		&handler.EnqueueRequestForObject{},
+		nsPredicate,
+	); err != nil {
+		return fmt.Errorf("failed to create watcher: %w", err)
+	}
+
+	return nil
+}
+
+func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	logger := r.log.With("kubermatic configuration", request.Name)
+	logger.Debug("Reconciling")
+
+	kc := &kubermaticv1.KubermaticConfiguration{}
+
+	if err := r.Get(ctx, request.NamespacedName, kc); err != nil {
+		if apierrors.IsNotFound(err) {
+			return reconcile.Result{}, nil
+		}
+
+		return reconcile.Result{}, fmt.Errorf("failed to get kubermatic configuration: %w", err)
+	}
+
+	if err := r.reconcile(ctx, logger, kc); err != nil {
+		r.recorder.Event(kc, corev1.EventTypeWarning, "ReconcilingFailed", err.Error())
+		return reconcile.Result{}, fmt.Errorf("failed to reconcile kubermatic configuration %s: %w", kc.Name, err)
+	}
+
+	return reconcile.Result{}, nil
+}
+
+func (r *Reconciler) reconcile(ctx context.Context, log *zap.SugaredLogger, kc *kubermaticv1.KubermaticConfiguration) error {
+	if kc.DeletionTimestamp != nil {
+		return nil
+	}
+	versions := r.versions
+	return kubermaticv1helper.UpdateKubermaticConfigurationStatus(ctx, r, kc, func(config *kubermaticv1.KubermaticConfiguration) {
+		config.Status.KubermaticEdition = versions.KubermaticCommit
+		config.Status.KubermaticVersion = versions.KubermaticEdition.String()
```

Comment:
Do we really want to store "Community Edition" / "Enterprise Edition" here, or would it make more sense to store some identifier-like value, like "CE"/"EE" here?